src/controllers/pages/MatrixChat.js

```diff
@@ -127,6 +127,31 @@ module.exports = {
                 });
                 this.notifyNewScreen('login');
                 break;
+            case 'cas_login':
+                if (this.state.logged_in) return;
+
+                var self = this;
+                var client = MatrixClientPeg.get();
+                var splitLocation = window.location.href.split('/');
+                var serviceUrl = splitLocation[0] + "//" + splitLocation[2];
```

Comment:
This bit looks suspiciously similar to the PR on vector-web. Can we factor this logic out, possibly by passing the `serviceUrl` as a `payload.param` rather than munging it directly out of the URL bar?